### PR TITLE
Added Dilation2D, Dilation2DBackpropFilter, Dilation2DBackpropInput

### PR DESCRIPTION
**Pull request by yangw1234**
## What changes were proposed in this pull request?

Added Dilation2D, Dilation2DBackpropFilter, Dilation2DBackpropInput

## How was this patch tested?

added unit tests

### CODE REVIEW

**Comment by yangw1234** [comment]
pr validation pass

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Dilation2DBackpropFilter.scala [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+import scala.reflect.ClassTag
+
+class Dilation2DBackpropFilter[T: ClassTag, D: ClassTag](
+         strides: Seq[Int],
+         rates: Seq[Int],
+         padding: String)(implicit ev: TensorNumeric[T], ev2: TensorNumeric[D])
+  extends Operation[Table, Tensor[D], T]{
+
+  private def getOutputSize(inputSize: Int, filterSize: Int, stride: Int, padding: String) = {
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Dilation2DBackpropFilter.scala]: can we add this function to a util class, so dialation and dialtionbackprop can share this?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Dilation2DBackpropFilter.scala [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+        val outputSize = (inputSize + stride - 1) / stride
+        val paddingNeeded = math.max(0, (outputSize - 1) * stride + filterSize - inputSize)
+        val padBefore = paddingNeeded / 2
+        val padAfter = paddingNeeded - padBefore
+        (outputSize, padBefore, padAfter)
+    }
+  }
+
+  private def dilation2DBackpropFilterFloat(
+                                         input: Tensor[Float],
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Dilation2DBackpropFilter.scala]: indent

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Dilation2DBackpropInput.scala [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+
+import scala.reflect.ClassTag
+
+class Dilation2DBackpropInput[T: ClassTag, D: ClassTag](strides: Seq[Int],
+                                                        rates: Seq[Int],
+                                                        padding: String)
+         (implicit ev: TensorNumeric[T], ev2: TensorNumeric[D])
+  extends Operation[Table, Tensor[D], T]{
+
+  private def getOutputSize(inputSize: Int, filterSize: Int, stride: Int, padding: String) = {
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Dilation2DBackpropInput.scala]: add it to util class?

**Comment by wzhongyuan** [comment]
@yangw1234   do you have serialization unit tests?

**Comment by yiheng** [comment]
Looks good to me. Please merge after PR validation

**Comment by yangw1234** [comment]
Added serialization tests. pr validation pass

**Pull request merged**